tests/waku_relay/test_protocol.nim

```diff
@@ -0,0 +1,1303 @@
+        otherTopic8 == pubsubTopicC
+        otherMsg8 == fromAnotherNodeWakuMessage6
+        anotherTopic7 == pubsubTopicB
+        anotherMsg7 == fromAnotherNodeWakuMessage5
+        anotherTopic8 == pubsubTopicC
+        anotherMsg8 == fromAnotherNodeWakuMessage6
+
+      # Finally stop the other nodes
+      await allFutures(otherSwitch.stop(), otherNode.stop(), anotherSwitch.stop(), anotherNode.stop())
+
+  suite "Unsubscribe":
+    asyncTest "Without Subscription":
+      # Given an external topic handler
+      let 
+        otherSwitch = newTestSwitch()
+        otherNode = await newTestWakuRelay(otherSwitch)
+      await allFutures(otherSwitch.start(), otherNode.start())
+      let otherTopicHandler: TopicHandler = otherNode.subscribe(pubsubTopic, simpleFutureHandler)
+      
+      # Given a node without a subscription
+      check:
+        node.subscribedTopics == []
+
+      # When unsubscribing from a pubsub topic
+      node.unsubscribe(pubsubTopic, otherTopicHandler)
+      # discard unsubscribe(node, pubsubTopic, otherTopicHandler)
+
+      # Then the node is still not subscribed
+      check:
+        node.subscribedTopics == []
+      
+      # Finally stop the other node
+      await allFutures(otherSwitch.stop(), otherNode.stop())
+
+    asyncTest "Single Node with Single Pubsub Topic":
+      # Given a node subscribed to a pubsub topic
+      let topicHandler = node.subscribe(pubsubTopic, simpleFutureHandler)
+      check node.subscribedTopics == pubsubTopicSeq
+
+      # When unsubscribing from the pubsub topic
+      node.unsubscribe(pubsubTopic, topicHandler)
+
+      # Then the node is not subscribed anymore
+      check node.subscribedTopics == []
+
+    asyncTest "Single Node with Multiple Pubsub Topics":
+      # Given other pubsub topic 
+      let pubsubTopicB = "pubsub-topic-b"
+
+      # Given a node subscribed to multiple pubsub topics
+      let 
+        topicHandler = node.subscribe(pubsubTopic, simpleFutureHandler)
+        topicHandlerB = node.subscribe(pubsubTopicB, simpleFutureHandler)
+      check node.subscribedTopics == @[pubsubTopic, pubsubTopicB]
+
+      # When unsubscribing from one of the pubsub topics
+      node.unsubscribe(pubsubTopic, topicHandler)
+      
+      # Then the node is still subscribed to the other pubsub topic
+      check node.subscribedTopics == @[pubsubTopicB]
+
+      # When unsubscribing from the other pubsub topic
+      node.unsubscribe(pubsubTopicB, topicHandlerB)
+
+      # Then the node is not subscribed anymore
+      check node.subscribedTopics == []
+
+  suite "Unsubscribe All":
+    asyncTest "Without subscriptions":
+      # Given a node without subscriptions
+      check node.subscribedTopics == []
+
+      # When unsubscribing from all pubsub topics
+      node.unsubscribeAll(pubsubTopic)
+      
+      # Then the node is still not subscribed
+      check node.subscribedTopics == []
+
+    asyncTest "Single Node with Single Pubsub Topic":
```

Comment:
Is this a duplicate of the one on L737?

Reply:
Nope, one tests `unsubscribe` while the other `unsubscribeAll`.